pyresample/utils/cf.py

```diff
@@ -64,15 +64,15 @@
 def _convert_XY_CF_to_Proj(crs, axis_info):
     """Convert XY values from CF to PROJ convention. With CF =< 1.9 only affects geostrationary projection."""
     crs_dict = crs.to_dict()
-    if crs_dict['proj'] == 'geos':
+    if (crs_dict['proj'] == 'geos') and (axis_info.get('unit') or 'radians' == 'radians'):
         # for geostationary projection, the values stored as x/y in CF are not directly
         #  the x/y along the projection axes, but are rather the scanning angles from
         #  the satellite. We must multiply them by the height of the satellite.
         satellite_height = crs_dict['h']
         for k in ('first', 'last', 'spacing'):
             axis_info[k] *= satellite_height
         # the unit is now the default (meters)
-        axis_info['units'] = None
+        axis_info['units'] = 'meters'
```

Comment:
Just thought of this, does pyproj/PROJ refer to this as `metre` or `m`? This is assuming that pyproj's CRS object is where we get the original version of this.

Reply:
Thanks for the review. It should be `m`. I will fix it and add the test later.